a: int = 5

func crazy()

    mostra a # expect 5
    
     a : int = 4
     b : int
     c : int
     d : int

    mostra a # expect 4


    se verdadeiro entao
        
        mostra a # expect 4
         a : int =  3
         b : int
         c : int
         d : int

        mostra a # expect 3


        func nested_crazy()
            
            mostra a # expect 3
             a : int = 2
            mostra a # expect 2

        fim

        nested_crazy()
    fim

fim
crazy()

#[output]:Erro: Funções locais não têm acesso ao escopo envolvente.
